deltacat/compute/compactor_v2/utils/task_options.py

```diff
@@ -106,30 +119,33 @@ def _get_merge_task_options(
                 str(hb_idx)
             ]
             for entry_index in range(entry_start, entry_end):
-                entry = compacted_delta_manifest.entries[entry_index]
+                entry: ManifestEntry = compacted_delta_manifest.entries[entry_index]
 
-                current_entry_size = estimate_manifest_entry_size_bytes(
+                current_entry_size: Optional[
+                    float
+                ] = estimate_manifest_entry_size_bytes(
                     entry=entry,
                     operation_type=OperationType.PYARROW_DOWNLOAD,
                     estimate_resources_params=estimate_resources_params,
                 )
-                current_entry_rows = estimate_manifest_entry_num_rows(
+                current_entry_rows: Optional[int] = estimate_manifest_entry_num_rows(
                     entry=entry,
                     operation_type=OperationType.PYARROW_DOWNLOAD,
                     estimate_resources_params=estimate_resources_params,
                 )
-
-                data_size += current_entry_size
-                num_rows += current_entry_rows
-
+                if current_entry_size:
+                    data_size += current_entry_size
+                if current_entry_rows:
+                    num_rows += current_entry_rows
                 if primary_keys:
-                    pk_size = estimate_manifest_entry_column_size_bytes(
+                    pk_size: Optional[
+                        float
+                    ] = estimate_manifest_entry_column_size_bytes(
                         entry=entry,
                         columns=primary_keys,
                         operation_type=OperationType.PYARROW_DOWNLOAD,
                         estimate_resources_params=estimate_resources_params,
                     )
-
                     if pk_size is None:
                         pk_size_bytes += current_entry_size
```

Comment:
If `current_entry_size` could be None, shouldn't we check for that here as well?

Reply:
Good catch